packages/web/pages/stake/index.tsx

```diff
@@ -71,33 +87,111 @@ export const Staking: React.FC = observer(() => {
     return { currency: osmo, amount: primitiveAmount.amount, denom: osmo };
   }, [osmo, primitiveAmount]);
 
+  const delegationQuery = cosmosQueries.queryDelegations.getQueryBech32Address(
+    account?.address ?? ""
+  );
+
+  const userValidatorDelegations = delegationQuery.delegations;
+
+  const usersValidatorsMap = useMemo(() => {
+    const delegationsMap = new Map<string, StakingType.Delegation>();
+
+    userValidatorDelegations.forEach((delegation: StakingType.Delegation) => {
+      delegationsMap.set(delegation.delegation.validator_address, delegation);
+    });
+
+    return delegationsMap;
+  }, [userValidatorDelegations]);
+
+  const amountDefault = getAmountDefault(amountConfig.fraction);
+  const amount = amountConfig.amount || "0";
+  const amountUSD = priceStore
+    .calculatePrice(
+      new CoinPretty(
+        osmo,
+        new Dec(amountConfig.amount || 0).mul(
+          DecUtils.getTenExponentNInPrecisionRange(osmo.coinDecimals)
+        )
+      )
+    )
```

Comment:
```suggestion
    .calculatePrice(
      new CoinPretty(
        osmo,
        amountConfig.getAmountPrimitive().amount
      )
    )
```

What you have works, nice! The decimals need to be removed from the amount before passing to CoinPretty. What I was trying to say (my bad for using wrong function name) was that the amount config has the decimal conversion within it.

Reply:
( I realized I selected too many lines so the suggestion is not directly usable )